homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,633 @@
+                        portmap,
+                        i,
+                        con_type,
+                    )
+                )
+
+    return entities_list
+
+
+async def async_profile_entities_list(
+    fritzbox_tools: FritzBoxTools, device_friendly_name: str
+) -> list[FritzBoxProfileSwitch]:
+    """Get list of profile entities."""
+    _LOGGER.debug("Setting up %s switches", SWITCH_TYPE_DEVICEPROFILE)
+    entities_list: list = []
+    if len(fritzbox_tools.fritz_profiles) > 0:
+        for profile in fritzbox_tools.fritz_profiles.keys():
+            entities_list.append(
+                FritzBoxProfileSwitch(fritzbox_tools, device_friendly_name, profile)
+            )
+    return entities_list
+
+
+async def async_wifi_entities_list(
+    fritzbox_tools: FritzBoxTools, device_friendly_name: str
+) -> list[FritzBoxWifiSwitch]:
+    """Get list of wifi entities."""
+    _LOGGER.debug("Setting up %s switches", SWITCH_TYPE_WIFINETWORK)
+    entities_list: list = []
+    std_table = {"ac": "5Ghz", "n": "2.4Ghz"}
+    networks: dict = {}
+    for i in range(4):
+        if ("WLANConfiguration" + str(i)) in fritzbox_tools.connection.services:
+            network_info = await async_service_call_action(
+                fritzbox_tools, "WLANConfiguration", str(i), "GetInfo"
+            )
+            if network_info:
+                ssid = network_info["NewSSID"]
+                if ssid in networks.values():
+                    networks[i] = f'{ssid} {std_table[network_info["NewStandard"]]}'
+                else:
+                    networks[i] = ssid
+
+    for net in networks:
+        entities_list.append(
+            FritzBoxWifiSwitch(fritzbox_tools, device_friendly_name, net, networks[net])
+        )
+
+    return entities_list
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up entry."""
+    _LOGGER.debug("Setting up switches")
+    fritzbox_tools: FritzBoxTools = hass.data[DOMAIN][entry.entry_id]
+
+    _LOGGER.debug("Fritzbox services: %s", fritzbox_tools.connection.services)
+
+    entities_list: list[
+        FritzBoxDeflectionSwitch
+        | FritzBoxPortSwitch
+        | FritzBoxProfileSwitch
+        | FritzBoxWifiSwitch
+    ] = []
+
+    entities_list.extend(
+        await async_deflection_entities_list(fritzbox_tools, entry.title)
+    )
+    entities_list.extend(await async_port_entities_list(fritzbox_tools, entry.title))
+    entities_list.extend(await async_profile_entities_list(fritzbox_tools, entry.title))
+    entities_list.extend(await async_wifi_entities_list(fritzbox_tools, entry.title))
+
+    async_add_entities(entities_list)
+
+
+class FritzBoxBaseSwitch:
+    """Fritz switch base class."""
+
+    def __init__(self, fritzbox_tools: FritzBoxTools, switch_info: SwitchInfo) -> None:
+        """Init Fritzbox port switch."""
+        self.fritzbox_tools: FritzBoxTools = fritzbox_tools
```

Comment:
```suggestion
        __super__.init(fritzbox_tools, f"{self._friendly_name} {self._description}")
```

Then change `self.fritzbox_tools` to `self._fritzbox_tools` everwhere here. and `self._name` to self._device_name`